stdlib/LinearAlgebra/test/cholesky.jl

```diff
@@ -392,4 +392,14 @@ end
 
 end
 
+@testset "issue #33704, casting low-rank CholeskyPivoted to Matrix" begin
+    T = Float64
+    n = 8
+    tol = 1e-8
+    A = randn(T, (n,1))
+    B = A*A'
+    C = cholesky(B, Val(true), tol = tol, check=false)
+    @test maximum(abs, B-Matrix(C)) < tol
```

Comment:
```suggestion
    @test B ≈ Matrix(C)
```
and then you can delete the `tol` definition above.

Reply:
Great, code is updated now.